src/sentry/relay/config/metric_extraction.py

```diff
@@ -390,7 +391,25 @@ def _convert_aggregate_and_query_to_metric(
             groupbys=groupbys,
         )
 
+        # TODO: switch to validate_rule_condition
+        validate_sampling_condition(json.dumps(on_demand_spec.condition))
+
         return on_demand_spec.query_hash, on_demand_spec.to_metric_spec(project)
+
+    except ValueError:
+        # raised by validate_sampling_condition
+        logger.error(
```

Comment:
Could you also please add a Datadog metric? (e.g. `on_demand_metrics.invalid_metric_spec`)

Reply:
`on_demand_metrics.errors.invalid_metric_spec`

Reply:
sure [242d025](https://github.com/getsentry/sentry/pull/59894/commits/242d0258d4fb2639ec43daa1e358d95326fc1fb4)